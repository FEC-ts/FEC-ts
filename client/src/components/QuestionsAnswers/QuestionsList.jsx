import React, {useState} from 'react';
import QuestionsListEntry from './QuestionsListEntry.jsx';
import {MoreQuestionsButton, AddQuestionButton} from './QuestionsButtons.jsx';
import QuestionModal from './QuestionModal.jsx';

const QuestionsList = ({questions, productInfo}) => {
  // console.log('this is productInfo', productInfo);
  // console.log('these are questions', questions);
  const [displayMoreQuestionsButton, setDisplayMoreQuestionsButton] = useState(true);
  const [questionCount, setQuestionCount] = useState(2);
  const [showQuestionModal, setShowQuestionModal] = useState(false);

  let displayedQuestions = [];

  if (questions.results) {
    displayedQuestions = questions.results.slice(0, questionCount);
  }

  const incrementCount = () => {
    let temp = questionCount + 2;
    setQuestionCount(temp);
    if (questions.results.length < temp) {
      setDisplayMoreQuestionsButton(!displayMoreQuestionsButton);
    }
  };

  const changeShowQuestionModal = () => {
    setShowQuestionModal(!showQuestionModal);
  };

  // const decrementCount = () => {
  //   setDisplayMoreQuestionsButton(!displayMoreQuestionsButton);
  //   setQuestionCount(2);
  // };

  if (displayedQuestions.length !== 0) {
    return (
      <div className="questions-list">
        {displayedQuestions.map((question, index) => <QuestionsListEntry key={index} question={question} productInfo={productInfo} />)}
        {(displayMoreQuestionsButton && questions.results.length > 2) && (
          <MoreQuestionsButton incrementCount={incrementCount} />
        )}
        {(!showQuestionModal) && (
          <AddQuestionButton changeShowQuestionModal={changeShowQuestionModal} />
        )}
        {(showQuestionModal) && (
          <QuestionModal productInfo={productInfo} changeShowQuestionModal={changeShowQuestionModal} />
        )}
      </div>
    )
  }

  return (
    <div>
      <h4>Q:</h4>
      <p>No Questions Asked Yet!</p>
      <button type="button" onClick={() => changeShowQuestionModal()}>Submit New Question</button>
      {(showQuestionModal) && (
        <QuestionModal productInfo={productInfo} changeShowQuestionModal={changeShowQuestionModal} />
      )}
    </div>
  )
};

export default QuestionsList;