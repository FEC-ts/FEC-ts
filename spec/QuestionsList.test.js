/** @jest-environment jsdom */
import renderer from 'react-test-renderer';
import QuestionsList from '../client/src/components/QuestionsAnswers/QuestionsList.jsx';
import {cleanup, fireEvent, render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

const testData = require('./QuestionStubs.js');
const user = userEvent.setup();

describe ('The question list should render elements', () => {
  it ('questions list should exist on the DOM', () => {
    const component = renderer.create(
      <QuestionsList questions={testData.questions} productInfo={testData.productInfo}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it ('expect the "More Answered Questions" button to exist when more than 2 questions are present', () => {
    const list = render(
      <QuestionsList questions={testData.questions} productInfo={testData.productInfo}/>
    );

    expect(list.queryByRole('button', {name: 'More Answered Questions'})).toBeDefined();
  });
});

// describe ('The answer modal should render proper elements', () => {
//   it ('answer modal should exist on DOM', () => {
//     const component = renderer.create(
//       <AnswerModal questions={testData.questions} productInfo={testData.productInfo}/>
//     );

//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });