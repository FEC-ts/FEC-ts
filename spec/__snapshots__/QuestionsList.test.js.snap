// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The questions list should render elements questions list should exist on the DOM 1`] = `
<div
  className="questions-list"
>
  <div>
    <div
      className="question-list-body"
    >
      <p
        className="question-body"
      >
        <span
          className="question-heading"
        >
          Q: 
        </span>
         
        How does this fit?
      </p>
      <p
        className="question-actions"
      >
        <span>
          <b>
            Helpful?
          </b>
        </span>
         
        <span>
          <a
            className="q-links"
            onClick={[Function]}
          >
            Yes (
            14
            )
          </a>
           
          <b
            className="q-pipes"
          >
            |
          </b>
        </span>
         
        <a
          className="q-links"
          onClick={[Function]}
        >
          Add Answer
        </a>
         
        <b
          className="q-pipes"
        >
          |
        </b>
         
        <a
          className="q-links"
          onClick={[Function]}
        >
          Report Question
        </a>
      </p>
    </div>
    <div
      className="answer-title"
    >
      <h4
        className="answer-heading"
      >
        A: 
      </h4>
       
      <div
        className="answer-list"
      >
        <p
          className="answer-body"
        >
          No Answers Yet!
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="question-list-body"
    >
      <p
        className="question-body"
      >
        <span
          className="question-heading"
        >
          Q: 
        </span>
         
        Will my heir try to usurp the throne if they have these shoes
      </p>
      <p
        className="question-actions"
      >
        <span>
          <b>
            Helpful?
          </b>
        </span>
         
        <span>
          <a
            className="q-links"
            onClick={[Function]}
          >
            Yes (
            1
            )
          </a>
           
          <b
            className="q-pipes"
          >
            |
          </b>
        </span>
         
        <a
          className="q-links"
          onClick={[Function]}
        >
          Add Answer
        </a>
         
        <b
          className="q-pipes"
        >
          |
        </b>
         
        <a
          className="q-links"
          onClick={[Function]}
        >
          Report Question
        </a>
      </p>
    </div>
    <div
      className="answer-title"
    >
      <h4
        className="answer-heading"
      >
        A: 
      </h4>
       
      <div
        className="answer-list"
      >
        <p
          className="answer-body"
        >
          No Answers Yet!
        </p>
      </div>
    </div>
  </div>
  <button
    className="more-questions-button"
    onClick={[Function]}
    type="button"
  >
    More Questions
  </button>
  <button
    className="add-questions-button"
    onClick={[Function]}
    type="button"
  >
    Add A Question +
  </button>
</div>
`;
